File: frontend/src/types/project.ts
export interface ProjectMember {
  id: string;
  name?: string | null;
  email: string;
}

export interface Project {
  id: string;
  name: string;
  client_name: string;
  status: string;
  compiled_briefing: string | null;
  briefing_token_count: number | null;
  briefing_was_truncated: boolean | null;
  briefing_state_version: number | null;
  briefing_priority_order: string[] | null;
  monthly_budget_usd: number | null;
  created_at: string;
  updated_at: string;
  created_by: string;
  document_count?: number;
  open_task_count?: number | null;
  members?: ProjectMember[];
}

/**
 * Membership row (join between user and project).
 * Renamed from the previous `ProjectMember` to avoid collision with the
 * lightweight member shape returned by `GET /api/projects`.
 */
export interface ProjectMembership {
  id: string;
  project_id: string;
  user_id: string;
  role: string;
  joined_at: string;
}

export interface User {
  id: string;
  email: string;
  name: string | null;
  created_at: string;
}

File: backend/app/schemas/project.py
import uuid
from datetime import datetime
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel


class ProjectCreate(BaseModel):
    name: str
    client_name: str = ""


class ProjectUpdate(BaseModel):
    name: str | None = None
    client_name: str | None = None
    status: str | None = None
    custom_instructions: str | None = None
    monthly_budget_usd: Decimal | None = None


class ProjectMemberOut(BaseModel):
    id: str
    name: str | None = None
    email: str


class StaleNoticeOut(BaseModel):
    is_stale: bool = True
    days_since_activity: int | None = None
    overdue_deadline_count: int = 0
    dismissed: bool = False
    text_de: str = ""
    text_en: str = ""


class ProjectResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    name: str
    client_name: str
    status: str
    compiled_briefing: str | None
    briefing_token_count: Optional[int] = None
    briefing_was_truncated: Optional[bool] = None
    briefing_state_version: Optional[int] = None
    briefing_priority_order: Optional[list[str]] = None
    custom_instructions: str | None = None
    monthly_budget_usd: Optional[Decimal] = None
    archived_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
    created_by: uuid.UUID
    document_count: int = 0
    open_task_count: int | None = None
    failed_document_count: int = 0
    unread_change_count: int = 0
    last_activity_at: datetime | None = None
    stale_marker: bool = False
    stale_notice: "StaleNoticeOut | None" = None
    members: list[ProjectMemberOut] = []


class ProjectMemberResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    project_id: uuid.UUID
    user_id: uuid.UUID
    role: str
    joined_at: datetime


class AddMemberRequest(BaseModel):
    user_id: uuid.UUID
    role: str = "editor"
